#!/bin/bash

cd ~/maternal_loading/2.public_data

# # 18
# RNAseq_SE_exon_step.sh   liver_male_rep1 1,2 25 danRer11_2 SRR5119935.fastq.gz ensGene > bin/log/liver_male_rep1.log 2>&1
# RNAseq_SE_exon_step.sh   liver_male_rep2 1,2 25 danRer11_2 SRR5119936.fastq.gz ensGene > bin/log/liver_male_rep2.log 2>&1
# RNAseq_SE_exon_step.sh   liver_male_rep3 1,2 25 danRer11_2 SRR5119937.fastq.gz ensGene > bin/log/liver_male_rep3.log 2>&1
# RNAseq_SE_exon_step.sh liver_female_rep1 1,2 25 danRer11_2 SRR5119938.fastq.gz ensGene > bin/log/liver_female_rep1.log 2>&1
# RNAseq_SE_exon_step.sh liver_female_rep2 1,2 25 danRer11_2 SRR5119939.fastq.gz ensGene > bin/log/liver_female_rep2.log 2>&1
# RNAseq_SE_exon_step.sh liver_female_rep3 1,2 25 danRer11_2 SRR5119940.fastq.gz ensGene > bin/log/liver_female_rep3.log 2>&1
# # 19
# RNAseq_PE_exon_step.sh liver_rep1 1,2 25 danRer11_2 SRR9022957_1.fastq.gz SRR9022957_2.fastq.gz ensGene > bin/log/liver_rep1.log 2>&1
# RNAseq_PE_exon_step.sh liver_rep2 1,2 25 danRer11_2 SRR9022958_1.fastq.gz SRR9022958_2.fastq.gz ensGene > bin/log/liver_rep2.log 2>&1
# RNAseq_PE_exon_step.sh liver_rep3 1,2 25 danRer11_2 SRR9022959_1.fastq.gz SRR9022959_2.fastq.gz ensGene > bin/log/liver_rep3.log 2>&1
# # 20
# RNAseq_PE_exon_step.sh valve_rep1 1,2 25 danRer11_2 SRR10060610_1.fastq.gz SRR10060610_2.fastq.gz ensGene > bin/log/valve_rep1.log 2>&1
# RNAseq_PE_exon_step.sh valve_rep2 1,2 25 danRer11_2 SRR10060611_1.fastq.gz SRR10060611_2.fastq.gz ensGene > bin/log/valve_rep2.log 2>&1
# RNAseq_PE_exon_step.sh valve_rep3 1,2 25 danRer11_2 SRR10060612_1.fastq.gz SRR10060612_2.fastq.gz ensGene > bin/log/valve_rep3.log 2>&1
# # 21
# RNAseq_SE_exon_step.sh retina_muller_rep1 1,2 25 danRer11_2 SRR9189638.fastq.gz ensGene > bin/log/retina_muller_rep1.log 2>&1
# RNAseq_SE_exon_step.sh retina_muller_rep2 1,2 25 danRer11_2 SRR9189639.fastq.gz ensGene > bin/log/retina_muller_rep2.log 2>&1
# RNAseq_SE_exon_step.sh retina_muller_rep3 1,2 25 danRer11_2 SRR9189640.fastq.gz ensGene > bin/log/retina_muller_rep3.log 2>&1
# # 22
# RNAseq_PE_exon_step.sh kidney_rep1 1,2 25 danRer11_2 SRR7659031_1.fastq.gz SRR7659031_2.fastq.gz ensGene > bin/log/kidney_rep1.log 2>&1
# RNAseq_PE_exon_step.sh kidney_rep2 1,2 25 danRer11_2 SRR7659032_1.fastq.gz SRR7659032_2.fastq.gz ensGene > bin/log/kidney_rep2.log 2>&1
# RNAseq_PE_exon_step.sh kidney_rep3 1,2 25 danRer11_2 SRR7659033_1.fastq.gz SRR7659033_2.fastq.gz ensGene > bin/log/kidney_rep3.log 2>&1
# RNAseq_PE_exon_step.sh kidney_rep4 1,2 25 danRer11_2 SRR7659034_1.fastq.gz SRR7659034_2.fastq.gz ensGene > bin/log/kidney_rep4.log 2>&1
# # 23
# RNAseq_PE_exon_step.sh kidney_HSC_gata2a_p_runx1_p 1,2 25 danRer11_2 SRR9320538_1.fastq.gz SRR9320538_2.fastq.gz ensGene > bin/log/kidney_HSC_gata2a_p_runx1_p.log 2>&1
# RNAseq_PE_exon_step.sh kidney_HSC_gata2a_n_runx1_p 1,2 25 danRer11_2 SRR9320539_1.fastq.gz SRR9320539_2.fastq.gz ensGene > bin/log/kidney_HSC_gata2a_n_runx1_p.log 2>&1
# RNAseq_PE_exon_step.sh kidney_HSC_gata2a_p_runx1_n 1,2 25 danRer11_2 SRR9320540_1.fastq.gz SRR9320540_2.fastq.gz ensGene > bin/log/kidney_HSC_gata2a_p_runx1_n.log 2>&1
# 24
RNAseq_PE_exon_step.sh 4dpf_vascular_muscle_acta2_p_rep1 1,2 25 danRer11_2 SRR7813744_1.fastq.gz SRR7813744_2.fastq.gz ensGene > bin/log/4dpf_vascular_muscle_acta2_p_rep1.log 2>&1
# 25
RNAseq_PE_exon_step.sh ventricle_rep1 1,2 25 danRer11_2 SRR9336405_1.fastq.gz SRR9336405_2.fastq.gz ensGene > bin/log/ventricle_rep1.log 2>&1
# 33
RNAseq_PE_exon_step.sh   epidermis_Neutrophils_rep3 1,2 25 danRer11_2 SRR8591749_1.fastq.gz SRR8591749_2.fastq.gz ensGene > bin/log/epidermis_Neutrophils_rep3.log 2>&1


### view in IGV
cd ~/maternal_loading/2.public_data/1_mapping
# sort & index
samtools sort -@ 20 -o 4dpf_vascular_muscle_acta2_p_rep1_sorted.bam 4dpf_vascular_muscle_acta2_p_rep1.bam && samtools index -@ 20 4dpf_vascular_muscle_acta2_p_rep1_sorted.bam
samtools sort -@ 20 -o ventricle_rep1_sorted.bam                    ventricle_rep1.bam                    && samtools index -@ 20 ventricle_rep1_sorted.bam
samtools sort -@ 20 -o epidermis_Neutrophils_rep3_sorted.bam        epidermis_Neutrophils_rep3.bam        && samtools index -@ 20 epidermis_Neutrophils_rep3_sorted.bam

# select the reads leading to segmentation fault
samtools view 4dpf_vascular_muscle_acta2_p_rep1_sorted.bam "chr13:17815141-17815289" > 4dpf_vascular_muscle_acta2_p_rep1_sorted_segmentFaultRead.sam
samtools view ventricle_rep1_sorted.bam                    "chr3:36688886-36689032"  > ventricle_rep1_sorted_segmentFaultRead.sam
samtools view epidermis_Neutrophils_rep3_sorted.bam        "chr3:2685196-2719996"    > epidermis_Neutrophils_rep3_sorted_segmentFaultRead.sam

cut -f 1 4dpf_vascular_muscle_acta2_p_rep1_sorted_segmentFaultRead.sam | sort | uniq > 4dpf_vascular_muscle_acta2_p_rep1_sorted_segmentFaultRead.list.txt
cut -f 1 ventricle_rep1_sorted_segmentFaultRead.sam | sort | uniq > ventricle_rep1_sorted_segmentFaultRead.list.txt # SRR9336405.22070825
cut -f 1 epidermis_Neutrophils_rep3_sorted_segmentFaultRead.sam | sort | uniq > epidermis_Neutrophils_rep3_sorted_segmentFaultRead.list.txt 

# exclude the error reads
# java -jar ~/software/picard.jar FilterSamReads I=4dpf_vascular_muscle_acta2_p_rep1_sorted.bam O=4dpf_vascular_muscle_acta2_p_rep1_sorted_filtered.bam READ_LIST_FILE=4dpf_vascular_muscle_acta2_p_rep1_sorted_segmentFaultRead.list.txt FILTER=excludeReadList
# java -jar ~/software/picard.jar FilterSamReads I=ventricle_rep1_sorted.bam O=ventricle_rep1_sorted_filtered.bam READ_LIST_FILE=ventricle_rep1_sorted_segmentFaultRead.list.txt FILTER=excludeReadList
java -jar ~/software/picard.jar FilterSamReads I=epidermis_Neutrophils_rep3_sorted.bam O=epidermis_Neutrophils_rep3_sorted_filtered.bam READ_LIST_FILE=epidermis_Neutrophils_rep3_sorted_segmentFaultRead.list.txt FILTER=excludeReadList

# re-run stringtie
cd ~/maternal_loading/2.public_data/2_expression_value_ExonUniq
# stringtie ../1_mapping/4dpf_vascular_muscle_acta2_p_rep1_sorted_filtered.bam -p 30 -G /mnt/Storage/home/wangyiman/source/bySpecies/danRer11_2/ensGene/danRer11_2.ensGene.ExonUniq.gtf -l 4dpf_vascular_muscle_acta2_p_rep1 -A RNA_seq_4dpf_vascular_muscle_acta2_p_rep1_ensGene_coverage.ExonUniq.txt -o RNA_seq_4dpf_vascular_muscle_acta2_p_rep1_ensGene_coverage.ExonUniq.gtf -e -B -v > ../bin/log/4dpf_vascular_muscle_acta2_p_rep1_rerun.log 2>&1
# stringtie ../1_mapping/ventricle_rep1_sorted_filtered.bam -p 30 -G /mnt/Storage/home/wangyiman/source/bySpecies/danRer11_2/ensGene/danRer11_2.ensGene.ExonUniq.gtf -l ventricle_rep1 -A RNA_seq_ventricle_rep1_ensGene_coverage.ExonUniq.txt -o RNA_seq_ventricle_rep1_ensGene_coverage.ExonUniq.gtf -e -B -v > ../bin/log/ventricle_rep1_rerun.log 2>&1
stringtie ../1_mapping/epidermis_Neutrophils_rep3_sorted_filtered.bam -p 30 -G /mnt/Storage/home/wangyiman/source/bySpecies/danRer11_2/ensGene/danRer11_2.ensGene.ExonUniq.gtf -l epidermis_Neutrophils_rep3 -A RNA_seq_epidermis_Neutrophils_rep3_ensGene_coverage.ExonUniq.txt -o RNA_seq_epidermis_Neutrophils_rep3_ensGene_coverage.ExonUniq.gtf -e -B -v > ../bin/log/epidermis_Neutrophils_rep3_rerun.log 2>&1


function rerun_segment_fault_samples{
    name=$1
    segment_fault_site=$2

    cd ~/maternal_loading/2.public_data/1_mapping
    samtools sort -@ 20 -o ${name}_sorted.bam ${name}.bam && samtools index -@ 20 ${name}_sorted.bam
    samtools view ${name}_sorted.bam $site > ${name}_sorted_segmentFaultRead.sam
    cut -f 1 ${name}_sorted_segmentFaultRead.sam | sort | uniq > ${name}_sorted_segmentFaultRead.list.txt
    java -jar ~/software/picard.jar FilterSamReads I=${name}_sorted.bam O=${name}_sorted_filtered.bam READ_LIST_FILE=${name}_sorted_segmentFaultRead.list.txt FILTER=excludeReadList

    cd ~/maternal_loading/2.public_data/2_expression_value_ExonUniq
    stringtie ../1_mapping/${name}_sorted_filtered.bam -p 30 -G /mnt/Storage/home/wangyiman/source/bySpecies/danRer11_2/ensGene/danRer11_2.ensGene.ExonUniq.gtf -l ${name} -A RNA_seq_ventricle_rep1_ensGene_coverage.ExonUniq.txt -o RNA_seq_${name}_ensGene_coverage.ExonUniq.gtf -e -B -v > ../bin/log/${name}_rerun.log 2>&1

    tail ../bin/log/${name}_rerun.log
}

rerun_segment_fault_samples 4dpf_vascular_muscle_acta2_p_rep1 "chr13:17815141-17815289"
rerun_segment_fault_samples ventricle_rep1 "chr3:36688886-36689032"
rerun_segment_fault_samples epidermis_Neutrophils_rep3 "chr3:2685196-2719996"