#!/bin/bash
cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/2_expression_value
paste <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_oocyte_20200904_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - ) <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_oocyte_20200904-2_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_oocyte_20200703_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_oocyte_20200703-2_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_64_20200515_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_64_20200531_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_64_20190503_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_64_20190607_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_1k_20200515_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_1k_20200531_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_1k_20190607_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_1k_20200515_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_oblong_20200515_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_npm2a_oblong_20200531_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_oblong_20190607_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) | paste - <(gtf_to_table.py -l 9 -f transcript RNA_seq_wt_oblong_20200515_ensGene_coverage.gtf | grep -v '#' | cut -f 10,14 | sort -k1,1 - | cut -f 2 ) > ../analysis/TPM_mat.tmp
cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/analysis
cat <(echo transcript_id$'\t'npm2a_oocyte_rep1$'\t'npm2a_oocyte_rep2$'\t'wt_oocyte_rep1$'\t'wt_oocyte_rep2$'\t'npm2a_64_rep1$'\t'npm2a_64_rep2$'\t'wt_64_rep1$'\t'wt_64_rep2$'\t'npm2a_1k_rep1$'\t'npm2a_1k_rep2$'\t'wt_1k_rep1$'\t'wt_1k_rep2$'\t'npm2a_oblong_rep1$'\t'npm2a_oblong_rep2$'\t'wt_oblong_rep1$'\t'wt_oblong_rep2) ../analysis/TPM_mat.tmp > ../analysis/TPM_mat.tsv
