#!/bin/bash

cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/0_raw_data

# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-N2-1K_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-N2-256_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-N2-64_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-N2-oblong_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-wt-1K_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-wt-256_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-wt-512_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200515/TU-TL-RNA-wt-oblong_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200531/TU-TL-npm2a-1kRNA_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200531/TU-TL-npm2a-256RNA_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200531/TU-TL-npm2a-64RNA_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200531/TU-TL-npm2a-oblongRNA_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200703/6l-oocyte-1_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200703/6l-oocyte-2_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200703/WT-oocyte-1_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200703/WT-oocyte-2_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200904/Npm2a-oocyte-1_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200904/Npm2a-oocyte-2_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200908/Orc1-oocyte-1_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20200908/Orc1-oocyte-2_combined_R[12].fastq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/1K-mcm6l-RNA_FRRA190036002-1a/1K-mcm6l-RNA_FRRA190036002-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/1K-wt-RNA_FRRA190035997-1a/1K-wt-RNA_FRRA190035997-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/256-mcm6l-RNA_FRRA190036001-1a/256-mcm6l-RNA_FRRA190036001-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/256-wt-RNA_FRRA190035996-1a/256-wt-RNA_FRRA190035996-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/64-mcm6l-RNA_FRRA190036000-1a/64-mcm6l-RNA_FRRA190036000-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/64-wt-RNA_FRRA190035995-1a/64-wt-RNA_FRRA190035995-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/6h-mcm6l-RNA_FRRA190036004-1a/6h-mcm6l-RNA_FRRA190036004-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/6h-wt-RNA_FRRA190035999-1a/6h-wt-RNA_FRRA190035999-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/oblong-mcm6l-RNA_FRRA190036003-1a/oblong-mcm6l-RNA_FRRA190036003-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190530/oblong-wt-RNA_FRRA190035998-1a/oblong-wt-RNA_FRRA190035998-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/1K-mcm6l-2-RNA_FRRA190043067-1a/1K-mcm6l-2-RNA_FRRA190043067-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/1K-wt-2-RNA_FRRA190043063-1a/1K-wt-2-RNA_FRRA190043063-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/256-mcm6l-2-RNA_FRRA190043066-1a/256-mcm6l-2-RNA_FRRA190043066-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/256-wt-2-RNA_FRRA190043062-1a/256-wt-2-RNA_FRRA190043062-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/64-mcm6l-2-RNA_FRRA190043065-1a/64-mcm6l-2-RNA_FRRA190043065-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/64-wt-2-RNA_FRRA190043061-1a/64-wt-2-RNA_FRRA190043061-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/oblong-mcm6l-2-RNA_FRRA190043068-1a/oblong-mcm6l-2-RNA_FRRA190043068-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20190607/oblong-wt-2-RNA_FRRA190043064-1a/oblong-wt-2-RNA_FRRA190043064-1a_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/mcm3l_RRAS24339-V/mcm3l_RRAS24339-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/mcm6l-1_RRAS24340-V/mcm6l-1_RRAS24340-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/mcm6l-2_RRAS24341-V/mcm6l-2_RRAS24341-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/npm2a_RRAS24338-V/npm2a_RRAS24338-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/orc1-1_RRAS24342-V/orc1-1_RRAS24342-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/orc1-2_RRAS24343-V/orc1-2_RRAS24343-V_[12].fq.gz .
# ln -s /mnt/Storage/home/wangwen/project/maternal_KO/inhouse/RNA_seq/20181022/WT_RRAS24337-V/WT_RRAS24337-V_[12].fq.gz .


cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq
# $0 <name> <step,+> <processer> <genomeVersion> <R1.fastq.gz,+> <R2.fastq.gz,+> <ensGene|refGene>

# RNAseq_PE_step.sh       npm2a_1k_20200515 1,2,3,4 5 danRer11_2        TU-TL-RNA-N2-1K_combined_R1.fastq.gz TU-TL-RNA-N2-1K_combined_R2.fastq.gz       ensGene > bin/log/npm2a_1k_20200515.log 2>&1 &
# RNAseq_PE_step.sh      npm2a_256_20200515 1,2,3,4 5 danRer11_2       TU-TL-RNA-N2-256_combined_R1.fastq.gz TU-TL-RNA-N2-256_combined_R2.fastq.gz      ensGene > bin/log/npm2a_256_20200515.log 2>&1 &
# RNAseq_PE_step.sh       npm2a_64_20200515 1,2,3,4 5 danRer11_2        TU-TL-RNA-N2-64_combined_R1.fastq.gz TU-TL-RNA-N2-64_combined_R2.fastq.gz       ensGene > bin/log/npm2a_64_20200515.log 2>&1 &
# RNAseq_PE_step.sh   npm2a_oblong_20200515 1,2,3,4 5 danRer11_2    TU-TL-RNA-N2-oblong_combined_R1.fastq.gz TU-TL-RNA-N2-oblong_combined_R2.fastq.gz   ensGene > bin/log/npm2a_oblong_20200515.log 2>&1 &
# RNAseq_PE_step.sh          wt_1k_20200515 1,2,3,4 5 danRer11_2        TU-TL-RNA-wt-1K_combined_R1.fastq.gz TU-TL-RNA-wt-1K_combined_R2.fastq.gz       ensGene > bin/log/wt_1k_20200515.log 2>&1 &
# RNAseq_PE_step.sh         wt_256_20200515 1,2,3,4 5 danRer11_2       TU-TL-RNA-wt-256_combined_R1.fastq.gz TU-TL-RNA-wt-256_combined_R2.fastq.gz      ensGene > bin/log/wt_256_20200515.log 2>&1 &
# wait
# RNAseq_PE_step.sh         wt_512_20200515 1,2,3,4 5 danRer11_2       TU-TL-RNA-wt-512_combined_R1.fastq.gz TU-TL-RNA-wt-512_combined_R2.fastq.gz      ensGene > bin/log/wt_512_20200515.log 2>&1 &
# RNAseq_PE_step.sh      wt_oblong_20200515 1,2,3,4 5 danRer11_2    TU-TL-RNA-wt-oblong_combined_R1.fastq.gz TU-TL-RNA-wt-oblong_combined_R2.fastq.gz   ensGene > bin/log/wt_oblong_20200515.log 2>&1 &
# RNAseq_PE_step.sh       npm2a_1k_20200531 1,2,3,4 5 danRer11_2      TU-TL-npm2a-1kRNA_combined_R1.fastq.gz TU-TL-npm2a-1kRNA_combined_R2.fastq.gz     ensGene > bin/log/npm2a_1k_20200531.log 2>&1 &
# RNAseq_PE_step.sh      npm2a_256_20200531 1,2,3,4 5 danRer11_2     TU-TL-npm2a-256RNA_combined_R1.fastq.gz TU-TL-npm2a-256RNA_combined_R2.fastq.gz    ensGene > bin/log/npm2a_256_20200531.log 2>&1 &
# RNAseq_PE_step.sh       npm2a_64_20200531 1,2,3,4 5 danRer11_2      TU-TL-npm2a-64RNA_combined_R1.fastq.gz TU-TL-npm2a-64RNA_combined_R2.fastq.gz     ensGene > bin/log/npm2a_64_20200531.log 2>&1 &
# RNAseq_PE_step.sh   npm2a_oblong_20200531 1,2,3,4 5 danRer11_2  TU-TL-npm2a-oblongRNA_combined_R1.fastq.gz TU-TL-npm2a-oblongRNA_combined_R2.fastq.gz ensGene > bin/log/npm2a_oblong_20200531.log 2>&1 &
# wait

# RNAseq_PE_step.sh   mcm6l_oocyte_20200703 1,2,3,4 5 danRer11_2            6l-oocyte-1_combined_R1.fastq.gz 6l-oocyte-1_combined_R2.fastq.gz           ensGene > bin/log/mcm6l_oocyte_20200703.log 2>&1 &
# RNAseq_PE_step.sh mcm6l_oocyte_20200703-2 1,2,3,4 5 danRer11_2            6l-oocyte-2_combined_R1.fastq.gz 6l-oocyte-2_combined_R2.fastq.gz           ensGene > bin/log/mcm6l_oocyte_20200703-2.log 2>&1 &
# RNAseq_PE_step.sh      wt_oocyte_20200703 1,2,3,4 5 danRer11_2            WT-oocyte-1_combined_R1.fastq.gz WT-oocyte-1_combined_R2.fastq.gz           ensGene > bin/log/wt_oocyte_20200703.log 2>&1 &
# RNAseq_PE_step.sh    wt_oocyte_20200703-2 1,2,3,4 5 danRer11_2            WT-oocyte-2_combined_R1.fastq.gz WT-oocyte-2_combined_R2.fastq.gz           ensGene > bin/log/wt_oocyte_20200703-2.log 2>&1 &
# wait
# RNAseq_PE_step.sh   npm2a_oocyte_20200904 1,2,3,4 5 danRer11_2         Npm2a-oocyte-1_combined_R1.fastq.gz Npm2a-oocyte-1_combined_R2.fastq.gz        ensGene > bin/log/npm2a_oocyte_20200904.log 2>&1 &
# RNAseq_PE_step.sh npm2a_oocyte_20200904-2 1,2,3,4 5 danRer11_2         Npm2a-oocyte-2_combined_R1.fastq.gz Npm2a-oocyte-2_combined_R2.fastq.gz        ensGene > bin/log/npm2a_oocyte_20200904-2.log 2>&1 &
# RNAseq_PE_step.sh    orc1_oocyte_20200908 1,2,3,4 5 danRer11_2          Orc1-oocyte-1_combined_R1.fastq.gz Orc1-oocyte-1_combined_R2.fastq.gz         ensGene > bin/log/orc1_oocyte_20200908.log 2>&1 &
# RNAseq_PE_step.sh  orc1_oocyte_20200908-2 1,2,3,4 5 danRer11_2          Orc1-oocyte-2_combined_R1.fastq.gz Orc1-oocyte-2_combined_R2.fastq.gz         ensGene > bin/log/orc1_oocyte_20200908-2.log 2>&1 &
# wait

# RNAseq_PE_step.sh       mcm6l_1k_20190503 1,2,3,4 5 danRer11_2       1K-mcm6l-RNA_FRRA190036002-1a_1.fq.gz 1K-mcm6l-RNA_FRRA190036002-1a_2.fq.gz ensGene > bin/log/mcm6l_1k_20190503.log 2>&1
# RNAseq_PE_step.sh          wt_1k_20190503 1,2,3,4 5 danRer11_2          1K-wt-RNA_FRRA190035997-1a_1.fq.gz 1K-wt-RNA_FRRA190035997-1a_2.fq.gz ensGene > bin/log/wt_1k_20190503.log 2>&1
# RNAseq_PE_step.sh      mcm6l_256_20190503 1,2,3,4 5 danRer11_2      256-mcm6l-RNA_FRRA190036001-1a_1.fq.gz 256-mcm6l-RNA_FRRA190036001-1a_2.fq.gz ensGene > bin/log/mcm6l_256_20190503.log 2>&1
# RNAseq_PE_step.sh         wt_256_20190503 1,2,3,4 5 danRer11_2         256-wt-RNA_FRRA190035996-1a_1.fq.gz 256-wt-RNA_FRRA190035996-1a_2.fq.gz ensGene > bin/log/wt_256_20190503.log 2>&1
# wait
# RNAseq_PE_step.sh       mcm6l_64_20190503 1,2,3,4 5 danRer11_2       64-mcm6l-RNA_FRRA190036000-1a_1.fq.gz 64-mcm6l-RNA_FRRA190036000-1a_2.fq.gz ensGene > bin/log/mcm6l_64_20190503.log 2>&1
# RNAseq_PE_step.sh          wt_64_20190503 1,2,3,4 5 danRer11_2          64-wt-RNA_FRRA190035995-1a_1.fq.gz 64-wt-RNA_FRRA190035995-1a_2.fq.gz ensGene > bin/log/wt_64_20190503.log 2>&1
# RNAseq_PE_step.sh     mcm6l_6hpf_20190503 1,2,3,4 5 danRer11_2       6h-mcm6l-RNA_FRRA190036004-1a_1.fq.gz 6h-mcm6l-RNA_FRRA190036004-1a_2.fq.gz ensGene > bin/log/mcm6l_6hpf_20190503.log 2>&1
# RNAseq_PE_step.sh        wt_6hpf_20190503 1,2,3,4 5 danRer11_2          6h-wt-RNA_FRRA190035999-1a_1.fq.gz 6h-wt-RNA_FRRA190035999-1a_2.fq.gz ensGene > bin/log/wt_6hpf_20190503.log 2>&1
# wait
# RNAseq_PE_step.sh   mcm6l_oblong_20190503 1,2,3,4 5 danRer11_2   oblong-mcm6l-RNA_FRRA190036003-1a_1.fq.gz oblong-mcm6l-RNA_FRRA190036003-1a_2.fq.gz ensGene > bin/log/mcm6l_oblong_20190503.log 2>&1
# RNAseq_PE_step.sh      wt_oblong_20190503 1,2,3,4 5 danRer11_2      oblong-wt-RNA_FRRA190035998-1a_1.fq.gz oblong-wt-RNA_FRRA190035998-1a_2.fq.gz ensGene > bin/log/wt_oblong_20190503.log 2>&1
# RNAseq_PE_step.sh       mcm6l_1k_20190607 1,2,3,4 5 danRer11_2     1K-mcm6l-2-RNA_FRRA190043067-1a_1.fq.gz 1K-mcm6l-2-RNA_FRRA190043067-1a_2.fq.gz ensGene > bin/log/mcm6l_1k_20190607.log 2>&1
# RNAseq_PE_step.sh          wt_1k_20190607 1,2,3,4 5 danRer11_2        1K-wt-2-RNA_FRRA190043063-1a_1.fq.gz 1K-wt-2-RNA_FRRA190043063-1a_2.fq.gz ensGene > bin/log/wt_1k_20190607.log 2>&1
# wait
# RNAseq_PE_step.sh      mcm6l_256_20190607 1,2,3,4 5 danRer11_2    256-mcm6l-2-RNA_FRRA190043066-1a_1.fq.gz 256-mcm6l-2-RNA_FRRA190043066-1a_2.fq.gz ensGene > bin/log/mcm6l_256_20190607.log 2>&1
# RNAseq_PE_step.sh         wt_256_20190607 1,2,3,4 5 danRer11_2       256-wt-2-RNA_FRRA190043062-1a_1.fq.gz 256-wt-2-RNA_FRRA190043062-1a_2.fq.gz ensGene > bin/log/wt_256_20190607.log 2>&1
# RNAseq_PE_step.sh       mcm6l_64_20190607 1,2,3,4 5 danRer11_2     64-mcm6l-2-RNA_FRRA190043065-1a_1.fq.gz 64-mcm6l-2-RNA_FRRA190043065-1a_2.fq.gz ensGene > bin/log/mcm6l_64_20190607.log 2>&1
# RNAseq_PE_step.sh          wt_64_20190607 1,2,3,4 5 danRer11_2        64-wt-2-RNA_FRRA190043061-1a_1.fq.gz 64-wt-2-RNA_FRRA190043061-1a_2.fq.gz ensGene > bin/log/wt_64_20190607.log 2>&1
# wait
# RNAseq_PE_step.sh   mcm6l_oblong_20190607 1,2,3,4 5 danRer11_2 oblong-mcm6l-2-RNA_FRRA190043068-1a_1.fq.gz oblong-mcm6l-2-RNA_FRRA190043068-1a_2.fq.gz ensGene > bin/log/mcm6l_oblong_20190607.log 2>&1
# RNAseq_PE_step.sh      wt_oblong_20190607 1,2,3,4 5 danRer11_2    oblong-wt-2-RNA_FRRA190043064-1a_1.fq.gz oblong-wt-2-RNA_FRRA190043064-1a_2.fq.gz ensGene > bin/log/wt_oblong_20190607.log 2>&1
# RNAseq_PE_step.sh       mcm3l_1k_20181022 1,2,3,4 5 danRer11_2                   mcm3l_RRAS24339-V_1.fq.gz mcm3l_RRAS24339-V_2.fq.gz ensGene > bin/log/mcm3l_1k_20181022.log 2>&1
# RNAseq_PE_step.sh       mcm6l_1k_20181022 1,2,3,4 5 danRer11_2                 mcm6l-1_RRAS24340-V_1.fq.gz mcm6l-1_RRAS24340-V_2.fq.gz ensGene > bin/log/mcm6l_1k_20181022.log 2>&1
# wait
# RNAseq_PE_step.sh     mcm6l_1k_20181022-2 1,2,3,4 5 danRer11_2                 mcm6l-2_RRAS24341-V_1.fq.gz mcm6l-2_RRAS24341-V_2.fq.gz ensGene > bin/log/mcm6l_1k_20181022-2.log 2>&1
# RNAseq_PE_step.sh       npm2a_1k_20181022 1,2,3,4 5 danRer11_2                   npm2a_RRAS24338-V_1.fq.gz npm2a_RRAS24338-V_2.fq.gz ensGene > bin/log/npm2a_1k_20181022.log 2>&1
# RNAseq_PE_step.sh        orc1_1k_20181022 1,2,3,4 5 danRer11_2                  orc1-1_RRAS24342-V_1.fq.gz orc1-1_RRAS24342-V_2.fq.gz ensGene > bin/log/orc1_1k_20181022.log 2>&1
# RNAseq_PE_step.sh      orc1_1k_20181022-2 1,2,3,4 5 danRer11_2                  orc1-2_RRAS24343-V_1.fq.gz orc1-2_RRAS24343-V_2.fq.gz ensGene > bin/log/orc1_1k_20181022-2.log 2>&1
# RNAseq_PE_step.sh          wt_1k_20181022 1,2,3,4 5 danRer11_2                      WT_RRAS24337-V_1.fq.gz WT_RRAS24337-V_2.fq.gz ensGene > bin/log/wt_1k_20181022.log 2>&1
# wait



cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/2_expression_value

# gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 mcm6l_oocyte_20200703,mcm6l_oocyte_20200703-2         -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm6l_oocyte_ensGene.diff > ../bin/log/WT_VS_mcm6l_oocyte_ensGene.diff.log 2>&1 &
# gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 npm2a_oocyte_20200904-2,npm2a_oocyte_20200904         -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_oocyte_ensGene.diff > ../bin/log/WT_VS_npm2a_oocyte_ensGene.diff.log 2>&1 &
# gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 orc1_oocyte_20200908-2,orc1_oocyte_20200908           -suf _ensGene.read_cnt -o  ../diff/WT_VS_orc1_oocyte_ensGene.diff > ../bin/log/WT_VS_orc1_oocyte_ensGene.diff.log  2>&1 &
wait
# gfold diff -s1                                wt_64_20190503,wt_64_20190607 -s2 mcm6l_64_20190503,mcm6l_64_20190607                   -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm6l_64_ensGene.diff > ../bin/log/WT_VS_mcm6l_64_ensGene.diff.log 2>&1 &
# gfold diff -s1                                wt_64_20190503,wt_64_20190607 -s2 npm2a_64_20200515,npm2a_64_20200531                   -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_64_ensGene.diff > ../bin/log/WT_VS_npm2a_64_ensGene.diff.log 2>&1 &
wait
# gfold diff -s1              wt_256_20190503,wt_256_20190607,wt_256_20200515 -s2 mcm6l_256_20190503,mcm6l_256_20190607                 -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm6l_256_ensGene.diff > ../bin/log/WT_VS_mcm6l_256_ensGene.diff.log 2>&1 &
# gfold diff -s1              wt_256_20190503,wt_256_20190607,wt_256_20200515 -s2 npm2a_256_20200515,npm2a_256_20200531                 -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_256_ensGene.diff > ../bin/log/WT_VS_npm2a_256_ensGene.diff.log 2>&1 &
wait
gfold diff -s1  wt_1k_20181022,wt_1k_20190503,wt_1k_20190607,wt_1k_20200515 -s2 mcm6l_1k_20190503,mcm6l_1k_20181022,mcm6l_1k_20190607,mcm6l_1k_20181022-2 -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm6l_1k_ensGene.diff > ../bin/log/../diff/WT_VS_mcm6l_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20181022,wt_1k_20190503,wt_1k_20190607,wt_1k_20200515 -s2 mcm3l_1k_20181022                                                         -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm3l_1k_ensGene.diff > ../bin/log/../diff/WT_VS_mcm3l_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20181022,wt_1k_20190503,wt_1k_20190607,wt_1k_20200515 -s2 npm2a_1k_20200531,npm2a_1k_20200515,npm2a_1k_20181022                     -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_1k_ensGene.diff > ../bin/log/../diff/WT_VS_npm2a_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20181022,wt_1k_20190503,wt_1k_20190607,wt_1k_20200515 -s2 orc1_1k_20181022,orc1_1k_20181022-2                                       -suf _ensGene.read_cnt -o ../diff/WT_VS_orc1_1k_ensGene.diff  > ../bin/log/../diff/WT_VS_orc1_1k_ensGene.diff.log 2>&1 &
wait
gfold diff -s1     wt_oblong_20190503,wt_oblong_20190607,wt_oblong_20200515 -s2 mcm6l_oblong_20190607,mcm6l_oblong_20190503           -suf _ensGene.read_cnt -o ../diff/WT_VS_mcm6l_oblong_ensGene.diff > ../bin/log/../diff/WT_VS_mcm6l_oblong_ensGene.diff.log 2>&1 &
gfold diff -s1     wt_oblong_20190503,wt_oblong_20190607,wt_oblong_20200515 -s2 npm2a_oblong_20200515,npm2a_oblong_20200531           -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_oblong_ensGene.diff > ../bin/log/../diff/WT_VS_npm2a_oblong_ensGene.diff.log 2>&1 &
wait
gfold diff -s1                                             wt_6hpf_20190503 -s2 mcm6l_6hpf_20190503                                   -suf _ensGene.read_cnt -o ../diff/WT_VS_npm2a_oblong_ensGene.diff > ../bin/log/../diff/WT_VS_mcm6l_6hpf_ensGene.diff.log 2>&1 &
wait






# ---------- by transcript ----------


cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/2_expression_value
sort ../diff_byTrans/samples_gtf_prepDE.tsv > ../diff_byTrans/samples_gtf_prepDE.sorted.tsv
sample_list_fi=../diff_byTrans/samples_gtf_prepDE.sorted.tsv
prepDE.py3 -i $sample_list_fi -l 150 -g gene_count_matrix.csv -t transcript_count_matrix.csv

cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/diff_byTrans
cut -f 6,9,8 ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/2_expression_value/t_data.ctab | tail -n +2 | sort -k1,1 > EnsGene.len
sed 's/,/\t/g' ../2_expression_value/transcript_count_matrix.csv | tail -n +2 | sort -k1,1 > transcript_count_matrix.dos.tab
sed 's/\r$//' transcript_count_matrix.dos.tab > transcript_count_matrix.tab ### remove dos ending "/r"

paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $2, $48, "-"}}' > mcm3l_1k_20181022_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $3, $48, "-"}}' > mcm6l_1k_20181022-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $4, $48, "-"}}' > mcm6l_1k_20181022_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $5, $48, "-"}}' > mcm6l_1k_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $6, $48, "-"}}' > mcm6l_1k_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $7, $48, "-"}}' > mcm6l_256_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $8, $48, "-"}}' > mcm6l_256_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $9, $48, "-"}}' > mcm6l_64_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $10, $48, "-"}}' > mcm6l_64_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $11, $48, "-"}}' > mcm6l_6hpf_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $12, $48, "-"}}' > mcm6l_oblong_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $13, $48, "-"}}' > mcm6l_oblong_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $14, $48, "-"}}' > mcm6l_oocyte_20200703-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $15, $48, "-"}}' > mcm6l_oocyte_20200703_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $16, $48, "-"}}' > npm2a_1k_20181022_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $17, $48, "-"}}' > npm2a_1k_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $18, $48, "-"}}' > npm2a_1k_20200531_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $19, $48, "-"}}' > npm2a_256_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $20, $48, "-"}}' > npm2a_256_20200531_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $21, $48, "-"}}' > npm2a_64_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $22, $48, "-"}}' > npm2a_64_20200531_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $23, $48, "-"}}' > npm2a_oblong_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $24, $48, "-"}}' > npm2a_oblong_20200531_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $25, $48, "-"}}' > npm2a_oocyte_20200904-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $26, $48, "-"}}' > npm2a_oocyte_20200904_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $27, $48, "-"}}' > orc1_1k_20181022-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $28, $48, "-"}}' > orc1_1k_20181022_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $29, $48, "-"}}' > orc1_oocyte_20200908-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $30, $48, "-"}}' > orc1_oocyte_20200908_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $31, $48, "-"}}' > wt_1k_20181022_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $32, $48, "-"}}' > wt_1k_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $33, $48, "-"}}' > wt_1k_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $34, $48, "-"}}' > wt_1k_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $35, $48, "-"}}' > wt_256_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $36, $48, "-"}}' > wt_256_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $37, $48, "-"}}' > wt_256_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $38, $48, "-"}}' > wt_512_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $39, $48, "-"}}' > wt_64_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $40, $48, "-"}}' > wt_64_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $41, $48, "-"}}' > wt_6hpf_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $42, $48, "-"}}' > wt_oblong_20190503_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $43, $48, "-"}}' > wt_oblong_20190607_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $44, $48, "-"}}' > wt_oblong_20200515_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $45, $48, "-"}}' > wt_oocyte_20200703-2_ensGene.count
paste transcript_count_matrix.tab EnsGene.len | awk 'BEGIN{FS=OFS="\t"}{if($1==$47){print $1, $49, $46, $48, "-"}}' > wt_oocyte_20200703_ensGene.count

cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/diff_byTrans

gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 mcm6l_oocyte_20200703-2,mcm6l_oocyte_20200703                             -suf _ensGene.count -o ./WT_VS_mcm6l_oocyte_ensGene.diff > ../bin/log/WT_VS_mcm6l_oocyte_ensGene.diff.log 2>&1 &
gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 npm2a_oocyte_20200904,npm2a_oocyte_20200904-2                             -suf _ensGene.count -o ./WT_VS_npm2a_oocyte_ensGene.diff > ../bin/log/WT_VS_npm2a_oocyte_ensGene.diff.log 2>&1 &
gfold diff -s1                      wt_oocyte_20200703-2,wt_oocyte_20200703 -s2 orc1_oocyte_20200908,orc1_oocyte_20200908-2                               -suf _ensGene.count -o ./WT_VS_orc1_oocyte_ensGene.diff  > ../bin/log/WT_VS_orc1_oocyte_ensGene.diff.log 2>&1 &
gfold diff -s1                                wt_64_20190607,wt_64_20190503 -s2 mcm6l_64_20190607,mcm6l_64_20190503                                       -suf _ensGene.count -o ./WT_VS_mcm6l_64_ensGene.diff     > ../bin/log/WT_VS_mcm6l_64_ensGene.diff.log 2>&1 &
gfold diff -s1                                wt_64_20190607,wt_64_20190503 -s2 npm2a_64_20200531,npm2a_64_20200515                                       -suf _ensGene.count -o ./WT_VS_npm2a_64_ensGene.diff     > ../bin/log/WT_VS_npm2a_64_ensGene.diff.log 2>&1 &
gfold diff -s1              wt_256_20190607,wt_256_20190503,wt_256_20200515 -s2 mcm6l_256_20190607,mcm6l_256_20190503                                     -suf _ensGene.count -o ./WT_VS_mcm6l_256_ensGene.diff    > ../bin/log/WT_VS_mcm6l_256_ensGene.diff.log 2>&1 &
gfold diff -s1              wt_256_20190607,wt_256_20190503,wt_256_20200515 -s2 npm2a_256_20200515,npm2a_256_20200531                                     -suf _ensGene.count -o ./WT_VS_npm2a_256_ensGene.diff    > ../bin/log/WT_VS_npm2a_256_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20200515,wt_1k_20190503,wt_1k_20190607,wt_1k_20181022 -s2 mcm6l_1k_20190503,mcm6l_1k_20190607,mcm6l_1k_20181022,mcm6l_1k_20181022-2 -suf _ensGene.count -o ./WT_VS_mcm6l_1k_ensGene.diff     > ../bin/log/WT_VS_mcm6l_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20200515,wt_1k_20190503,wt_1k_20190607,wt_1k_20181022 -s2 mcm3l_1k_20181022                                                         -suf _ensGene.count -o ./WT_VS_mcm3l_1k_ensGene.diff     > ../bin/log/WT_VS_mcm3l_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20200515,wt_1k_20190503,wt_1k_20190607,wt_1k_20181022 -s2 npm2a_1k_20200515,npm2a_1k_20200531,npm2a_1k_20181022                     -suf _ensGene.count -o ./WT_VS_npm2a_1k_ensGene.diff     > ../bin/log/WT_VS_npm2a_1k_ensGene.diff.log 2>&1 &
gfold diff -s1  wt_1k_20200515,wt_1k_20190503,wt_1k_20190607,wt_1k_20181022 -s2 orc1_1k_20181022-2,orc1_1k_20181022                                       -suf _ensGene.count -o ./WT_VS_orc1_1k_ensGene.diff      > ../bin/log/WT_VS_orc1_1k_ensGene.diff.log 2>&1 &
gfold diff -s1     wt_oblong_20190503,wt_oblong_20200515,wt_oblong_20190607 -s2 mcm6l_oblong_20190503,mcm6l_oblong_20190607                               -suf _ensGene.count -o ./WT_VS_mcm6l_oblong_ensGene.diff > ../bin/log/WT_VS_mcm6l_oblong_ensGene.diff.log 2>&1 &
gfold diff -s1     wt_oblong_20190503,wt_oblong_20200515,wt_oblong_20190607 -s2 npm2a_oblong_20200515,npm2a_oblong_20200531                               -suf _ensGene.count -o ./WT_VS_npm2a_oblong_ensGene.diff > ../bin/log/WT_VS_npm2a_oblong_ensGene.diff.log 2>&1 &
gfold diff -s1                                             wt_6hpf_20190503 -s2 mcm6l_6hpf_20190503                                                       -suf _ensGene.count -o ./WT_VS_mcm6l_6hpf_ensGene.diff   > ../bin/log/WT_VS_mcm6l_6hpf_ensGene.diff.log 2>&1 &
wait

# cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/diff/by_transcript
# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2     mcm3l_1k.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm3l_1k.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm3l_1k.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm3l_1k.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2 mcm6l_oblong.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm6l_oblong.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm6l_oblong.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm6l_oblong.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2     npm2a_64.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > npm2a_64.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 npm2a_64.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > npm2a_64.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2 npm2a_oocyte.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > npm2a_oocyte.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 npm2a_oocyte.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > npm2a_oocyte.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2     mcm6l_1k.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm6l_1k.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm6l_1k.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm6l_1k.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2  orc1_oocyte.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > orc1_oocyte.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 orc1_oocyte.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > orc1_oocyte.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2      orc1_1k.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > orc1_1k.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 orc1_1k.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > orc1_1k.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2     mcm6l_64.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm6l_64.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm6l_64.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm6l_64.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2 npm2a_oblong.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > npm2a_oblong.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 npm2a_oblong.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > npm2a_oblong.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2 mcm6l_oocyte.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm6l_oocyte.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm6l_oocyte.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm6l_oocyte.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2    mcm6l_256.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > mcm6l_256.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 mcm6l_256.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > mcm6l_256.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2     npm2a_1k.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > npm2a_1k.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 npm2a_1k.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > npm2a_1k.genelist.tsv

# joinLANG=en_EN join -o 2.2 1.1 1.7 1.8 -1 1 -2 1 <(tail -n+2    npm2a_256.tab) <(cut -f 1,12 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7.ensGene.genePredExt | sort -k1,1) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4}' > npm2a_256.genelist.tmp
# joinLANG=en_EN join -o 2.2 1.1 1.2 1.3 1.4 -1 1 -2 1 <(sort -k1,1 npm2a_256.genelist.tmp) <(sort -k1,1 /mnt/Storage/home/wangwen/source/bySpecies/zv9/danRer7_ensGeneToGeneSymbol.txt) | awk 'BEGIN{FS=" ";OFS="\t"}{print $1, $2, $3, $4, $5}' > npm2a_256.genelist.tsv

# rm *.tmp




# cd ~/maternal_loading/1.private_data/genes4_ko_analysis/RNAseq/diff/by_transcript
# gfold diff -s1 wt_1k_20181022 -s2 mcm6l_1k_20181022 -suf _ensGene.count -o ./diff/WT_VS_mcm6l_1k_20181022_ensGene.diff &
# gfold diff -s1 wt_1k_20181022 -s2 mcm3l_1k_20181022 -suf _ensGene.count -o ./diff/WT_VS_mcm3l_1k_20181022_ensGene.diff &
# gfold diff -s1 wt_1k_20181022 -s2 npm2a_1k_20181022 -suf _ensGene.count -o ./diff/WT_VS_npm2a_1k_20181022_ensGene.diff &
# gfold diff -s1 wt_1k_20181022 -s2 orc1_1k_20181022-2,orc1_1k_20181022 -suf _ensGene.count -o ./diff/WT_VS_orc1_1k_20181022_ensGene.diff &
wait